### PR TITLE
Update to latest wgpu and egui/eframe

### PR DESCRIPTION
**Pull request by emilk**
### What
* Tracks egui `master` branch
* Tracks https://github.com/rerun-io/egui_tiles/pull/78

~This changes the interface between HTML/JS and WASM from passing a canvas id, to passing the canvas element directly (https://github.com/emilk/egui/pull/4780). This means the preview of this PR is broken, and merging this PR will likely break it for all subsequent PRs. @jprochazk do you have a good idea how we can fix this?~

### TODO
* [x] The `re_renderer` example needs a winit update
* [x] Fix uses of deprecated functions.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested `pixi run rerun-web`
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7218?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7218?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7218)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by Wumpf** [comment]
FYI I just landed https://github.com/emilk/egui/pull/4964
It already picked the right wgpu versions here though :)

File: crates/viewer/re_renderer_examples/framework.rs [hunk 1]
```diff
@@ -378,35 +346,78 @@ pub fn load_rerun_mesh(re_ctx: &RenderContext) -> Vec<re_renderer::renderer::Mes
...
         re_log::setup_logging();
 
+        let window = winit::window::WindowAttributes::default()
+            .with_title(format!("re_renderer sample - {}", E::title()))
+            .with_inner_size(winit::dpi::PhysicalSize {
+                width: 1920,
+                height: 1080,
+            });
+
+        // TODO(emilk): port this to the winit 0.30 API, using maybe https://docs.rs/winit/latest/winit/platform/web/trait.EventLoopExtWebSys.html ?
```
Comment (emilk) [inline crates/viewer/re_renderer_examples/framework.rs:417]: I gave up porting the wasm32 portion of the examples to latest winit

File: rerun_js/web-viewer/index.ts [hunk 2]
```diff
@@ -195,7 +195,7 @@ export class WebViewer {
 
     this.#handle = new WebHandle_class({ ...options, fullscreen });
     try {
-      await this.#handle.start(this.#canvas.id);
```
Comment (jprochazk) [inline rerun_js/web-viewer/index.ts:198]: We should also remove all the `this.#id` shenanigans here, seeing as it's no longer used

**Review by jprochazk — APPROVED** [review]
Web stuff LGTM

File: crates/utils/re_log/src/lib.rs [hunk 3]
```diff
@@ -73,9 +73,11 @@ const CRATES_AT_INFO_LEVEL: &[&str] = &[
     "h2",
     "hyper",
     "ureq",
-    // only let rustls run in debug mode: https://github.com/rerun-io/rerun/issues/3104
+    // only let rustls log in debug mode: https://github.com/rerun-io/rerun/issues/3104
     #[cfg(debug_assertions)]
     "rustls",
+    // winit 0.30.5 spams about `set_cursor_visible` calls. It's gone on winit master, so hopefully gone in next winit release.
+    "winit",
```
Comment (abey79) [inline crates/utils/re_log/src/lib.rs:80]: add a `TODO`/issue to track that?

File: crates/viewer/re_chunk_store_ui/src/chunk_store_ui.rs [hunk 4]
```diff
@@ -326,30 +326,28 @@ impl DatastoreUi {
...
-                        .body(|body| {
-                            body.rows(
-                                re_ui::DesignTokens::table_line_height(),
-                                chunks.len(),
-                                row_ui,
-                            );
-                        });
-                });
+                let table_builder = egui_extras::TableBuilder::new(ui)
+                    .id_source(chunk_store.id())
```
Comment (abey79) [inline crates/viewer/re_chunk_store_ui/src/chunk_store_ui.rs:330]: niiice

**Review by abey79 — APPROVED** [review]
Looks good!

File: crates/utils/re_log/src/lib.rs (hunk 3 above)
Comment (emilk) [reply crates/utils/re_log/src/lib.rs:80]: We don't really have a system in place for tracking new releases… this is a very minor thing anyways (we rarely care about debug logs from winit)

**Pull request merged**